src/backends/pytorch/torch_backend.cc

```diff
@@ -0,0 +1,687 @@
+}
+
+Status
+LibTorchBackend::CreateExecutionContext(
+    const std::string& instance_name, const int gpu_device,
+    const std::unordered_map<std::string, std::string>& paths)
+{
+  cudaError_t cuerr;
+
+  // For a GPU context, determine the model file to use for device
+  // compute capability. CPU always uses the default model file.
+  std::string cc;
+  std::string cc_model_filename;
+  if (gpu_device == Context::NO_GPU_DEVICE) {
+    cc_model_filename = Config().default_model_filename();
+    LOG_INFO << "Creating instance " << instance_name << " on CPU using "
+             << cc_model_filename;
+  } else {
+    cudaDeviceProp cuprops;
+    cuerr = cudaGetDeviceProperties(&cuprops, gpu_device);
+    if (cuerr != cudaSuccess) {
+      return Status(
+          RequestStatusCode::INTERNAL,
+          "unable to get CUDA device properties for " + Name() + ": " +
+              cudaGetErrorString(cuerr));
+    }
+
+    cc = std::to_string(cuprops.major) + "." + std::to_string(cuprops.minor);
+    const auto& cc_itr = Config().cc_model_filenames().find(cc);
+    cc_model_filename = (cc_itr == Config().cc_model_filenames().end())
+                            ? Config().default_model_filename()
+                            : cc_itr->second;
+  }
+
+  const auto& lp_itr = paths.find(cc_model_filename);
+  if (lp_itr == paths.end()) {
+    return Status(
+        RequestStatusCode::INTERNAL, "unable to find LibTorch model '" +
+                                         cc_model_filename + "' for " + Name());
+  }
+
+  if (gpu_device == Context::NO_GPU_DEVICE) {
+    LOG_INFO << "Creating instance " << instance_name << " on CPU using "
+             << cc_model_filename;
+  } else {
+    LOG_INFO << "Creating instance " << instance_name << " on GPU "
+             << gpu_device << " (" << cc << ") using " << cc_model_filename;
+  }
+
+  // Max batch size. A value of 0 in the config becomes NO_BATCHING.
+  const int mbs = (Config().max_batch_size() <= 0) ? Context::NO_BATCHING
+                                                   : Config().max_batch_size();
+
+  contexts_.emplace_back(instance_name, gpu_device, mbs);
+  Context* context = contexts_.back().get();
+
+  // If this is a sequence model then add the required inputs...
+  if (Config().has_sequence_batching()) {
+    RETURN_IF_ERROR(ValidateSequenceControl(
+        ModelSequenceBatching::Control::CONTROL_SEQUENCE_START, &input_names));
+    RETURN_IF_ERROR(ValidateSequenceControl(
+        ModelSequenceBatching::Control::CONTROL_SEQUENCE_READY, &input_names));
+  }
+
+  if (gpu_device == Context::NO_GPU_DEVICE) {
+    device_ = torch::Device(torch::kCPU);
+  } else {
+    device_ = torch::Device(torch::kCUDA, device_option.device_id)
+  }
+  try {
+    // lp_itr->second is the torch model path
+    torch_model_ = torch::jit::load(lp_itr->second, device_);
+    model_name_= Config().name();
+    max_batch_size_ = Config().max_batch_size();
+    gpu_device_ = gpu_device;
```

Comment:
Aren't these variables members under Context struct? Should prefix them with `context->...`